libgnucash/engine/Account.cpp

```diff
@@ -3428,6 +3428,24 @@ xaccAccountGetNoclosingBalanceAsOfDate (Account *acc, time64 date)
     return GetBalanceAsOfDate (acc, date, TRUE);
 }
 
+gnc_numeric
+xaccAccountGetReconciledBalanceAsOfDate (Account *acc, time64 date)
+{
+    gnc_numeric balance = gnc_numeric_zero();
+
+    g_return_val_if_fail(GNC_IS_ACCOUNT(acc), gnc_numeric_zero());
+
+    for (GList *node = GET_PRIVATE(acc)->splits; node; node = node->next)
+    {
+        Split *split = (Split*) node->data;
+        if (xaccSplitGetReconcile (split) != YREC) continue;
+        if (xaccSplitGetDateReconciled (split) > date) continue;
```

Comment:
Any reason not to combine these into a single `if` with `||` between the conditions?

Reply:
No. I'll do before merge. It was just a stylistic choice, to reduce indentation. Similar to:
```scheme
(cond
 ((fail-condition-1) #f)
 ((fail-condition-2) #f)
 (else (do-actual work)))
```

Reply:
The amended code will test for pass-conditions instead of fail-conditions instead, which is even clearer.